### PR TITLE
refactor [PROD-13063]: refactor connector commands with api service

### PR DESCRIPTION
**Pull request by vysahjk**

### CODE REVIEW

File: Babylon/commands/api/workspaces/service/api.py [hunk 1]
```diff
@@ -70,11 +70,35 @@ def update(self):
...
+            workspace_key=self.state["api"]["workspace_key"],
+            name="eventhub",
+        )
+        if not secret_eventhub:
+            logger.error("workspace secret key is missing in vault")
+            sys.exit(1)
+        details_json = {"dedicatedEventHubKey": secret_eventhub.replace('"', "")}
+        details_json = json.dumps(details_json, indent=4, default=str)
+        work_id = self.state["api"]["workspace_id"]
+        workspace_id = self.state["api"]["workspace_id"]
```
Comment (esasova) [inline Babylon/commands/api/workspaces/service/api.py:94]: why use two variables for the same value ? :thinking:

File: Babylon/commands/api/workspaces/send_key.py [hunk 2]
```diff
@@ -17,25 +19,23 @@
...
-                             azure_token,
-                             type="POST",
-                             data=details_json)
-    if response is None:
-        return CommandResponse.fail()
-    logger.info(f"Successfully update key in workspace {work_id}")
+    service_state = state["services"]
+    service_state["api"]["organization_id"] = (organization_id or service_state["api"]["organization_id"])
+    service_state["api"]["workspace_id"] = (workspace_id or service_state["api"]["workspace_id"])
+    service_state["api"]["workspace_key"] = (workspace_id or service_state["api"]["workspace_key"])
```
Comment (esasova) [inline Babylon/commands/api/workspaces/send_key.py:37]: It means that `workspace_id `can serve as a key, yes ?

**Review by esasova — APPROVED** [review]
Just questions, LGTM :+1:

**Review by Leopold-Cramer — APPROVED** [review]

File: Babylon/commands/api/workspaces/service/api.py (hunk 1 above)
Comment (vysahjk) [reply Babylon/commands/api/workspaces/service/api.py:94]: oups!

File: Babylon/commands/api/workspaces/send_key.py (hunk 2 above)
Comment (vysahjk) [reply Babylon/commands/api/workspaces/send_key.py:37]: my bad 👍🏽

**Pull request merged**